modules/services/pulseeffects.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.pulseeffects;
+
+  presetOpts = optionalString (cfg.preset != "") "--load-preset ${cfg.preset}";
+
+in {
+  meta.maintainers = [ maintainers.jonringer ];
+
+  options.services.pulseeffects = {
+    enable =
+      mkEnableOption "Pulseeffects daemon. Used for advanced audio mixing.";
+
+    preset = mkOption {
+      type = types.lines;
+      default = "";
+      defaultText = literalExample "";
+      description = ''
+        Which preset to use when starting pulseeffects.
+        Will likely need to launch pulseeffects to initially create preset.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # running pulseeffects will just attach itself to gapplication service
+    # at-spi2-core is to minimize journalctl noise of:
+    # "AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files"
+    home.packages = [ pkgs.pulseeffects pkgs.at-spi2-core ];
+
+    # Will need to add `services.dbus.packages = with pkgs; [ gnome3.dconf ];`
+    # to /etc/nixos/configuration.nix for daemon to work correctly
+
+    systemd.user = {
+      services.pulseeffects = {
```

Comment:
Shouldn't pulseeffects be triggered by some systemd target? Or is it dbus activated? Searched a little and it seems it needs a graphical environment? If so perhaps need to add
```
Unit.After = [ "graphical-session-pre.target" ];
Unit.PartOf = [ "graphical-session.target" ];
Install.WantedBy = [ "graphical-session.target" ];
```
?

Reply:
using the `--gapplication-service` option essentially starts this in a headless state. But normally (e.g. `pulseeffects`), it's required.

Reply:
>  Or is it dbus activated?

This is probably the case:
```
GApplication options
  --gapplication-service     Enter GApplication service mode (use from D-Bus service files)
```

Reply:
updated it to `After = [ "dbus.service" ];`

Reply:
The unit section was important... fixed

Reply:
made it into `After = [ "dbus.service" "graphical-sessions-pre.target" ];`

as I was also getting:
```
Apr 22 19:01:01 jon-desktop systemd[2202]: Started Pulseeffects daemon.
Apr 22 19:01:01 jon-desktop .pulseeffects-w[10018]: cannot open display: :0
Apr 22 19:01:01 jon-desktop systemd[2202]: pulseeffects.service: Main process exited, code=exited, status=1/FAILURE
Apr 22 19:01:01 jon-desktop systemd[2202]: pulseeffects.service: Failed with result 'exit-code'.
Apr 22 19:01:01 jon-desktop systemd[2202]: pulseeffects.service: Scheduled restart job, restart counter is at 6.
Apr 22 19:01:01 jon-desktop systemd[2202]: Stopped Pulseeffects daemon.
Apr 22 19:01:01 jon-desktop systemd[2202]: pulseeffects.service: Start request repeated too quickly.
Apr 22 19:01:01 jon-desktop systemd[2202]: pulseeffects.service: Failed with result 'exit-code'.
Apr 22 19:01:01 jon-desktop systemd[2202]: Failed to start Pulseeffects daemon.
```